cloudify_libvirt/domain_tasks.py

```diff
@@ -44,47 +64,53 @@ def configure(**kwargs):
             'Failed to open connection to the hypervisor'
         )
 
-    domain_file = kwargs.get('domain_file')
-    domain_template = kwargs.get('domain_template')
+    template_params = _update_template_params(template_params)
 
-    if domain_file:
-        domain_template = ctx.get_resource(domain_file)
+    try:
+        if ctx.instance.runtime_properties.get("use_external_resource"):
+            # lookup the default domain by name
+            try:
+                dom = conn.lookupByName(
+                    ctx.instance.runtime_properties["resource_id"])
+            except Exception as e:
```

Comment:
@0lvin Cannot we be more specific here when catching the exception ? because down there you check if dom is `None` to raise an `NonRecoverableError`